src/FSharp.Build/Fsi.fs

```diff
@@ -161,6 +161,12 @@ type public Fsi() as this =
 
         builder
 
+    let outputWriter = new StringWriter()
```

Comment:
Is the lifetimescope aligned with life of the parent type? e.g. no reset needed?
What about memory capture?

I am wondering if the an object possibly huge shouldn't be either conditioned (tests, debug,..) or have a max. limit of content it will hold. Like a circular text bugger, but of course this should not rewrite its own.

Reply:
I think the lifetime of this is tied to the msbuild execution of the task (running a single script). It should be created / collected in the context of a single test case execution.

I was worried about modifying this at all, because my understanding of it is very low, but it seems this ToolTask is only used in a subset of FSharpSuite tests.

Reply:
The Fsi task itself is also used for "dotnet fsi", which could be used by people by long running (rich outputting) scenarios.

Reply:
The wrapper (defined in [FSharpTests.Directory.Build.targets](https://github.com/dotnet/fsharp/pull/17815/files/c26b39851380010b935194177d0047ae7a50e685#diff-04c19d53f7b43c8ca27f31e527990bd74e1ccf897929d3b858f81b52025653d8)) is just for tests in this repo.
But the Fsi build task goes into the SDK.

Reply:
Oh, thanks, I see. I'll find a way to do it to not modify the public stuff at all.

Reply:
What if `<RunFSharpScript>` passed a signalling flag to `<Fsi>` about this being a test scenario, and only then allowing to accumulate all (or some, via circular buffer) of output?

Reply:
I'm doing the OO by subtyping Fsi tool task :D, let's see how it goes.

Reply:
Should be good now.